crates/aptos/src/move_tool/mod.rs

```diff
@@ -6,50 +6,122 @@
 //! TODO: Examples
 //!
 
-use crate::CliResult;
+use crate::{
+    common::{types::MovePackageDir, utils::to_common_result},
+    CliResult, Error,
+};
+use aptos_vm::natives::aptos_natives;
 use clap::{Parser, Subcommand};
-use move_core_types::errmap::ErrorMapping;
-use move_vm_types::gas_schedule::INITIAL_COST_SCHEDULE;
-
-pub mod chain;
+use move_binary_format::CompiledModule;
+use move_cli::package::cli::{run_move_unit_tests, UnitTestResult};
+use move_package::BuildConfig;
+use move_unit_test::UnitTestingConfig;
+use std::{collections::HashSet, path::Path};
 
 /// CLI tool for performing Move tasks
 ///
 #[derive(Subcommand)]
 pub enum MoveTool {
-    Command(MoveCli),
+    Compile(CompilePackage),
+    Test(TestPackage),
 }
 
 impl MoveTool {
     pub async fn execute(self) -> CliResult {
         match self {
-            // TODO: Rethink using the Move CLI and think about how we can make the experience better
-            MoveTool::Command(tool) => tool.execute(),
+            MoveTool::Compile(tool) => to_common_result(tool.execute().await),
+            MoveTool::Test(tool) => to_common_result(tool.execute().await),
         }
     }
 }
 
+/// Compiles a package and returns the [`ModuleId`]s
+#[derive(Parser)]
+pub struct CompilePackage {
```

Comment:
what about output?

Reply:
The output is saved to it's default position of `build/` within the package folder.  Could make this configurable